### ISSUE TITLE
Gaussian Linear Blur effect broken

### ISSUE THREAD

**Issue by YorVeX** [Xaymar/obs-StreamFX#822]
### Operating System

Windows 10 1903 and higher

### OBS Studio Version?

27.1

### StreamFX Version

0.11.1

### OBS Studio Log

Really doesn't matter

### Current Behavior

Adding a Blur effect of "Gaussian Linear" type does nothing, original output stays unchanged.

### Expected Behavior

Blur effect being applied.

### Steps to Reproduce the Bug

Add "Blur" filter to e.g. an image source, set the type to "Guassian Linear", try various settings like "Subtype" or "Size", they all do nothing. "Gaussian" in comparison works fine.

### Any additional Information we need to know?

I tried my previous version StreamFX 0.9.3.0 as a comparison and the filter works fine there. When I go back to 0.11.1 it stops working again.

As an additional test I copied the data\obs-plugins\StreamFX\effects\blur\gaussian-linear.effect over from the StreamFX 0.9.3 package. Interestingly this is working, but needs a higher "Size" settings to achieve the same blur level.

**Comment by Xaymar** [comment]
"Linear" variants will become the default in the future, and have been phased out. There is a WIP complete rework of the blur filter. Please use Dual Filtering instead of Gaussian or Box blur as well.

**Comment by YorVeX** [comment]
The current description indicates that Dual Filtering doesn't fit all use cases though:
> Should be preferred if fine control over blur size is not required.

But I appreciate that hint, because actually all of my use cases so far could be covered by Dual Filtering even when controls are not as fine as for the other blur options - and if it's faster at the same time I prefer that of course.

**Comment by Xaymar** [comment]
Dual Filtering can only be controlled in 2^N steps, where you control N. After a size of 16px (4 in Dual Filtering), there is practically no practical need for other blur steps. Though it is possible to emulate this, and a lot of software out there does it - downscale the source with linear first to a fraction of the size but no less than 50%, then upscale after blur with linear to original size. 1/100th of the cost of actual Gaussian Blur, for a 95% accurate Gaussian blur approximation.

**Comment by GT500org** [comment]
Personally I find all of the built-in blur options to be way too blurry even on their lowest setting, so I use the gaussian blur shader for more fine grained control. It allows me to add only enough blurring to prevent compression artifacting, without adding more than necessary.

**Comment by YorVeX** [comment]
Yeah, after working through my OBS config I had also found two cases where I apply the blur effect to a Text source to smoothen the overall look after several other effects had been applied to it before. And there if I just go up one small notch with the blur the text just looks plain stupid, so there is no way to use Dual Filtering on that. But Xaymar said "Linear variants will become the default" and I wasusing Box filter there so I just switched to Box linear and hopefully I am fine now also for future StreamFX updates.
